src/System.IO.FileSystem/tests/Directory/GetFileSystemEntries_str.cs

```diff
@@ -3,271 +3,147 @@
-                    Directory.GetFileSystemEntries(strTempDir);
-                    iCountErrors++;
-                    printerr("Error_1003! Expected exception not thrown");
-                }
-                catch (ArgumentException)
-                {
-                }
-                catch (Exception exc)
-                {
-                    iCountErrors++;
-                    printerr("Error_1004! Unexpected exceptiont thrown: " + exc.ToString());
-                }
-            }
+    #region UniversalTests
 
-            //With lot's of \'s at the end
-            iCountTestcases++;
-            try
-            {
-                String strTempDir = Directory.GetCurrentDirectory() + new string(Path.DirectorySeparatorChar, 5);
-                strArr = Directory.GetFileSystemEntries(strTempDir);
-                if (strArr == null || strArr.Length == 0)
-                {
-                    printerr("Error_1234!!! INvalid number of file system entries count :: " + strArr.Length);
-                    iCountErrors++;
-                }
-            }
-            catch (Exception exc)
-            {
-                iCountErrors++;
-                printerr("Error_1006! Unexpected exceptiont thrown: " + exc.ToString());
-            }
+    [Fact]
+    public void NullFileName()
+    {
+        Assert.Throws<ArgumentNullException>(() => GetEntries(null));
+    }
 
-            //With the current directory
-            iCountTestcases++;
-            try
-            {
-                strArr = Directory.GetFileSystemEntries(s_strTFPath);
-                if (strArr == null || strArr.Length == 0)
-                {
-                    printerr("Error_2434!!! INvalid number of file system entries count :: " + strArr.Length);
-                    iCountErrors++;
-                }
-            }
-            catch (Exception exc)
-            {
-                iCountErrors++;
-                printerr("Error_12321!!! Unexpected exceptiont thrown: " + exc.ToString());
-            }
+    [Fact]
+    public void EmptyFileName()
+    {
+        Assert.Throws<ArgumentException>(() => GetEntries(""));
+    }
 
-            //With valid directory
-            dir2 = new DirectoryInfo(dirName);
-            dir2.Create();
+    [Fact]
+    public void NonexistentDirectory()
+    {
+        Assert.Throws<DirectoryNotFoundException>(() => GetEntries(GetTestFilePath()));
+    }
 
-            strArr = Directory.GetFileSystemEntries(dirName);
-            iCountTestcases++;
-            if (strArr.Length != 0)
-            {
-                iCountErrors++;
-                printerr("Error_207v7! Incorrect number of directories returned");
-            }
+    [Fact]
+    public void FileNameWithSpaces()
+    {
+        String strTempDir = "             ";
+        Assert.Throws<ArgumentException>(() => GetEntries(strTempDir));
```

Comment:
That's interesting! Since that is the case, I'll need to go through the tests again and fix this, as whitespace file/directory name checks are in essentially every test.